scripts/utils.js

```diff
@@ -12,17 +12,17 @@ module.exports = {
             .replace(/^&/, "and-")
             .replace(/&$/, "-and")
             .replace(/&/g, "-and-")
-            .replace(/[ !:’'°]/g, "")
-            .replace(/à|á|â|ã|ä/g, "a")
-            .replace(/ç|č|ć/g, "c")
-            .replace(/è|é|ê|ë/g, "e")
-            .replace(/ì|í|î|ï/g, "i")
-            .replace(/ñ|ň|ń/g, "n")
-            .replace(/ò|ó|ô|õ|ö/g, "o")
-            .replace(/š|ś/g, "s")
-            .replace(/ù|ú|û|ü/g, "u")
-            .replace(/ý|ÿ/g, "y")
-            .replace(/ž|ź/g, "z")
+            .replace(/đ/g, "d")
+            .replace(/ħ/g, "h")
+            .replace(/ı/g, "i")
+            .replace(/ĸ/g, "k")
+            .replace(/ŀ/g, "l")
+            .replace(/ł/g, "l")
+            .replace(/ß/g, "ss")
+            .replace(/ŧ/g, "t")
+            .normalize("NFD")
+            .replace(/[\u0300-\u036f]/g, "")
+            .replace(/[^\w\-]/g, "")
```

Comment:
I was a bit surprised that `_` is included in `\w` so I would personally prefer we use the RegExp below to be more explicit (note that the hyphen does not need to be escaped):

```suggestion
            .replace(/[^a-zA-Z0-9_-]/g, "")
```

But if there's consensus on using `\w` I'm happy with that as well, its behaviour is well documented after all.

Reply:
Yeah `[^a-z0-9_-]` (`A-Z` isn't needed) is probably better for clarity and easier updating. Updated. I always feels it's safer to escape the `-` here and makes it a bit clearer to people who may not be as familiar with JS.

While I'm at it, do you want me to dupe `normalizeSearchTerm` into `utils.js` and update `titleToFilename` accordingly so you can import `normalizeSearchTerm` from this repo for use on the site?

Reply:
> Yeah `[^a-z0-9_-]` (`A-Z` isn't needed) is probably better for clarity and easier updating. Updated. I always feels it's safer to escape the `-` here and makes it a bit clearer to people who may not be as familiar with JS.

Fair enough, I also only recently realized it's optional :upside_down_face: 


> While I'm at it, do you want me to dupe `normalizeSearchTerm` into `utils.js` and update `titleToFilename` accordingly so you can import `normalizeSearchTerm` from this repo for use on the new site?

I'm not sure if I follow. [`utils.js`](https://github.com/simple-icons/simple-icons/blob/develop/scripts/utils.js) is not exposed through the NPM package of simple-icons so that wouldn't help. Moreover, shouldn't `titleToFilename` and `normalizeSearchTerm` be the same function (perhaps one being the alias of the other for clarity)?

I think we should use your suggestion @PeterShaggyNoble and put the titleToFilename logic in a separate package so that it can easily be shared between repos. We just need to keep `normalizeSearchTerm` in this repo in sync for a bit until the new site is in use (*).

---

(*): because here we can't easily import JS modules on the website - without resorting to the latest browser features - while with the new repo that is easy.

Reply:
> Fair enough, I also only recently realized it's optional

In that case, this should be ready to go now, pending someone testing it.

> I'm not sure if I follow. utils.js is not exposed through the NPM package of simple-icons so that wouldn't help.

Ah, OK, I thought we might be able to pull it in for the new website with some dependency "magic"!

> I think we should use your suggestion @PeterShaggyNoble and put the titleToFilename logic in a separate package so that it can easily be shared between repos

If that what it takes 🙂 I'll leave that to someone with more GH savvy than I to do, separately to this PR.

> Ow and by the way, yes

Once this PR gets tested & merged, I'll submit one to the website's repo to keep the 2 scripts in sync for now.